rafay/data_cluster_blueprint_status.go

```diff
@@ -0,0 +1,573 @@
+				"addon_status_list": &schema.Schema{
+					Description: "cluster blueprint addon status",
+					Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+						"addon_name": &schema.Schema{
+							Description: "name of the blueprint addon",
+							Optional:    true,
+							Type:        schema.TypeString,
+						},
+						"namespace": &schema.Schema{
+							Description: "namespce of the blueprint addon",
+							Optional:    true,
+							Type:        schema.TypeString,
+						},
+						"publish_status": &schema.Schema{
+							Description: "status of the blueprint addon",
+							Optional:    true,
+							Type:        schema.TypeString,
+						},
+						"publish_time": &schema.Schema{
+							Description: "time of the blueprint addon publish",
+							Optional:    true,
+							Type:        schema.TypeString,
+						},
+						"conditions": &schema.Schema{
+							Description: "conditions blueprint addon status",
+							Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+								"type": &schema.Schema{
+									Description: "condition type of the blueprint addon",
+									Optional:    true,
+									Type:        schema.TypeString,
+								},
+								"status": &schema.Schema{
+									Description: "condition status of the blueprint addon",
+									Optional:    true,
+									Type:        schema.TypeString,
+								},
+								"last_updated": &schema.Schema{
+									Description: "condition updated time of the blueprint addon",
+									Optional:    true,
+									Type:        schema.TypeString,
+								},
+								"reason": &schema.Schema{
+									Description: "condition reason of the blueprint addon publish",
+									Optional:    true,
+									Type:        schema.TypeString,
+								},
+								"retrycount": &schema.Schema{
+									Description: "retry count of blueprint addon publish",
+									Optional:    true,
+									Type:        schema.TypeInt,
+								},
+								"max_retrycount": &schema.Schema{
+									Description: "maxium retry count of blueprint addon publish",
+									Optional:    true,
+									Type:        schema.TypeInt,
+								},
+								"components": &schema.Schema{
+									Description: "components of blueprint addon",
+									Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+										"name": &schema.Schema{
+											Description: "component name of the blueprint addon",
+											Optional:    true,
+											Type:        schema.TypeString,
+										},
+										"reason": &schema.Schema{
+											Description: "component failure reason of the blueprint addon publish",
+											Optional:    true,
+											Type:        schema.TypeString,
+										},
+										"failures": &schema.Schema{
+											Description: "component failures blueprint addon",
+											Elem: &schema.Resource{Schema: map[string]*schema.Schema{
+												"message": &schema.Schema{
+													Description: "component failure message of the blueprint addon",
```

Comment:
@stephan-rafay There are couple of other attributes added to `failure` object. Here is the rafay-common PR for this https://github.com/RafaySystems/rafay-common/pull/979/files
cc @saisrikar-rafay right?

Reply:
[Added reason & name field to failure